go.mod

```diff
@@ -26,101 +26,105 @@ require (
 	google.golang.org/protobuf v1.27.1
 	gopkg.in/resty.v1 v1.12.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.23.4
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
-	k8s.io/kubernetes v1.23.4
-	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704
-	sigs.k8s.io/controller-runtime v0.11.1
-)
-
-require (
-	github.com/apache/tinkerpop/gremlin-go v0.0.0-20220530191148-29272fa563ec
+	k8s.io/api v0.24.2
+	k8s.io/apimachinery v0.24.2
+	k8s.io/client-go v0.24.2
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+	sigs.k8s.io/controller-runtime v0.12.2
 	sigs.k8s.io/yaml v1.3.0
 )
 
 require (
-	cloud.google.com/go v0.93.3 // indirect
+	cloud.google.com/go v0.81.0 // indirect
```

Comment:
My assumption is that indirect dependencies are managed by `go mod`, not edited by us manually.
So after updating direct dependencies, I deleted all the indirect ones and re-ran `go mod tidy` to restart from a clean slate.
But this results in a few lower versions like this one, I'm not sure why.

I can revisit this if this wasn't the correct approach.

Reply:
The ``go mod`` commands are not "stable" in a sense that they would generate same output each time. I'm assuming they use map[] underneath for dependency graphs and resolve, in which case the iteration order is random and as such can generate different output.

Reply:
So do you think the discrepancy matters here? It looks like if the version is higher than what `tidy` would suggest, it's not modified. I can go back to the versions from `main` and re-run it.

Reply:
If I do that I get those differences:
```diff
-       cloud.google.com/go v0.81.0 // indirect
+       cloud.google.com/go v0.93.3 // indirect

-       github.com/Azure/go-autorest/autorest v0.11.18 // indirect
-       github.com/Azure/go-autorest/autorest/adal v0.9.13 // indirect
+       github.com/Azure/go-autorest/autorest v0.11.20 // indirect
+       github.com/Azure/go-autorest/autorest/adal v0.9.15 // indirect

-       github.com/form3tech-oss/jwt-go v3.2.3+incompatible // indirect

+       github.com/golang-jwt/jwt/v4 v4.0.0 // indirect

+       github.com/golang/snappy v0.0.4 // indirect

-       go.uber.org/atomic v1.7.0 // indirect
+       go.uber.org/atomic v1.9.0 // indirect
```

Reply:
I don't think it matters which one it picks as long as it compiles afterwards. Clearly there's two dependencies which have different version of certain module, so picking either one could break the other.